src/scorecard.js

```diff
@@ -0,0 +1,42 @@
+function Scorecard() {
+  this._frameScore = [];
+  this._frame = 1;
+  this._gameScore = [[]]
+}
+
+Scorecard.prototype.showScore = function() {
+  return this.sumFrameScore();
+}
+
+Scorecard.prototype.enterScore = function(pins) {
+  if (this.sumFrameScore() + pins > 10) throw "maximum input per frame is 10";
+  this._frameScore.push(pins);
+  if (this._gameScore[this._frame -1].includes(10)) {
+    this._gameScore[this._frame -1].push(pins);
+  }
+  if ((this._frameScore.length === 2) || (pins === 10)) {
+    this.frameReset();
+  }
+}
+
+Scorecard.prototype.isFrame = function() {
+  return this._frame;
+}
+
+Scorecard.prototype.frameReset = function() {
```

Comment:
I like that you moved this out as a separate method.

Reply:
Thanks!